### PR TITLE
[INLONG-9149][Agent] Add sender manager for file collect

### LINKED ISSUE #9149: [Improve][Agent] Add sender manager for file collect
### Description

Add sender manager for file collect

### InLong Component

InLong Agent

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by justinwwhuang**
### Prepare a Pull Request
[INLONG-9149][Agent] Add sender manager for file collect

- Fixes #9149 

### Motivation

Add sender manager for file collect

### Modifications

Add sender manager for file collect

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sinks/filecollect/SenderManager.java [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+
+    /**
+     * Send message to proxy by batch, use message cache.
+     */
+    private void sendBatchWithRetryCount(SenderMessage message, int retry) {
+        boolean suc = false;
+        while (!suc) {
+            try {
+                // suc = true;
+                // AgentSenderCallback cb = new AgentSenderCallback(message, retry);
```
Comment (doleyzi) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sinks/filecollect/SenderManager.java:313]: It is recommended to remove the commented code

**Review by doleyzi — APPROVED** [review]

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sinks/filecollect/SenderManager.java (hunk 1 above)
Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sinks/filecollect/SenderManager.java:313]: get

File: inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sinks/filecollect/SenderManager.java [hunk 2]
```diff
@@ -0,0 +1,454 @@
...
+        dims.put(KEY_INLONG_GROUP_ID, groupId);
+        dims.put(KEY_INLONG_STREAM_ID, streamId);
+        return getMetricItem(dims);
+    }
+
+    /**
+     * sender
+     *
+     * @param tagName group id
+     * @return DefaultMessageSender
```
Comment (EMsnap) [inline inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sinks/filecollect/SenderManager.java:226]: no return here and what's the meaning of tagname group id ?

Comment (justinwwhuang) [reply inlong-agent/agent-plugins/src/main/java/org/apache/inlong/agent/plugin/sinks/filecollect/SenderManager.java:226]: the annotation is out of date

**Review by EMsnap — APPROVED** [review]

**Review by dockerzhang — APPROVED** [review]

**Pull request merged**